### DISCUSSION TITLE
Vertical ThresholdLine

### DISCUSSION THREAD

**Discussion by PedroFrr** [patrykandpatrick/vico#615] [Ideas]
Hi all!

First of all thank you for the great library!! I'm using version **2.0.0-alpha.11** it with **Compose** without any hitch, everything's working flawlessly. 

By looking at the documentation I see that it's possible to add an horizontal ThresholdLine. But I'm wondering if it's possible to draw it on the X axis (vertical) instead? 

I've also tried using a guideline but I've notice that guidelines can only be drawn for the X axis entries values, so drawing if in between bars would not be possible. 

What I am trying to achieve is the yellow line as seen bellow:

![image](https://github.com/patrykandpatrick/vico/assets/29357833/709faafa-0558-4339-b241-f1038a83a888)

**Comment by patrickmichalik** [comment]
Hello! Thanks for the kind words. Vico doesn’t have such a component yet, but we’ll be looking into adding it—I’m moving this discussion to the “Ideas” category. In the meantime, you can implement such a line as a custom `Decoration`. [The `HorizontalLine` definition](https://github.com/patrykandpatrick/vico/blob/v2.0.0-alpha.12/vico/core/src/main/java/com/patrykandpatrick/vico/core/chart/decoration/HorizontalLine.kt#L49) can serve as a good starting point. (In [Vico 2.0.0 Alpha 12](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.12), `ThresholdLine` was split into two new APIs, `HorizontalBox` and `HorizontalLine`.) If you need any help with this, let me know.

Incidentally, if you create a custom `AxisItemPlacer.Horizontal` implementation, you can get a `HorizontalAxis` to display a guideline for any _x_ value. Here, however, a custom `Decoration` is the best option.

**Reply by PedroFrr** [reply]
Thank you very much for the answer! For the time being I went with the decoration approach and making a new VerticalLine copying all the things needed from HorizontalLine!

**Reply by patrickmichalik** [reply]
Sounds good. You’re welcome!
